File: crabzilla-vertx/src/main/java/crabzilla/vertx/ProjectionData.java
package crabzilla.vertx;

import crabzilla.DomainEvent;
import lombok.Value;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

@Value
public class ProjectionData implements Serializable {

  UUID uowId;
  Long uowSequence;
  String targetId;
  List<DomainEvent> events;

}

File: krabzilla-vertx/src/main/java/crabzilla/vertx/events/projection/ProjectionData.kt
package crabzilla.vertx.events.projection

import crabzilla.DomainEvent
import java.io.Serializable
import java.util.*

class ProjectionData(val uowId: UUID, val uowSequence: Long, val targetId: String, val events: List<DomainEvent>)
  : Serializable
